### PR TITLE
Use e2epod.CreateExecPodOrFail()

### PR DESCRIPTION
**Pull request by oomichi**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

The function is defined for running command with `kubectl exec` as [1].
The pods which were created with launchHostExecPod() run commands with
`kubectl exec` via framework.RunHostCmd(), so we don't need to maintain
local function and can just use e2epod.CreateExecPodOrFail() instead.

[1]: https://github.com/kubernetes/kubernetes/blob/d7e7136bbce0fa55361c0d6ebdc40357e4caf11d/test/e2e/framework/pod/resource.go#L431-L432

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/network/service.go [hunk 1]
```diff
@@ -1920,7 +1920,7 @@ var _ = SIGDescribe("Services", func() {
 		err = t.DeleteService(serviceName)
 		framework.ExpectNoError(err, "failed to delete service: %s in namespace: %s", serviceName, ns)
 
-		hostExec := launchHostExecPod(f.ClientSet, f.Namespace.Name, "hostexec")
+		hostExec := e2epod.CreateExecPodOrFail(f.ClientSet, f.Namespace.Name, "hostexec", nil)
```
Comment (johnbelamaric) [inline test/e2e/network/service.go]: This does not create a pod with host networking, whereas the previous code did. Same for below. What is the intended behavior?

Comment (oomichi) [reply test/e2e/network/service.go]: @johnbelamaric 
Thanks for your review.
That is a nice point I missed.
If applying this change, HostNetwork test cases become less, it is fine to keep as current code.

**Review by oomichi — COMMENTED** [review]
/close

**Pull request closed without merging**